iceoryx_posh/test/moduletests/test_posh_runtime.cpp

```diff
@@ -0,0 +1,320 @@
+}
+
+
+// test class creates instance of Poshruntime, so when getInstance() is called without name it reuturns extisting
+// instance
+TEST_F(PoshRuntime_test, DISABLED_AppnameEmpty)
+{
+    EXPECT_DEATH({ iox::runtime::PoshRuntime::getInstance(); },
+                 "Cannot initialize runtime. Application name has not been specified!");
+}
+
+
+TEST_F(PoshRuntime_test, GetInstanceName)
+{
+    const std::string appname = "/app";
+
+    auto& sut = PoshRuntime::getInstance(appname);
+
+    EXPECT_EQ(sut.getInstanceName(), appname);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareApplication_ReturnValue)
+{
+    uint32_t uniqueIdCounter = iox::popo::BasePortData::s_uniqueIdCounter;
+
+    const auto applicationPortData = m_senderRuntime->getMiddlewareApplication();
+
+    EXPECT_EQ(std::string("/sender"), applicationPortData->m_processName);
+    EXPECT_EQ(iox::capro::ServiceDescription(0u, 0u, 0u), applicationPortData->m_serviceDescription);
+    EXPECT_EQ(false, applicationPortData->m_toBeDestroyed);
+    EXPECT_EQ(uniqueIdCounter, applicationPortData->m_uniqueId);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareApplication_ApplicationlistOverflow)
+{
+    m_errorHandlerCalled = false;
+
+    for (auto i = 0u; i < iox::MAX_PROCESS_NUMBER; ++i)
+    {
+        m_senderRuntime->getMiddlewareApplication();
+    }
+
+    EXPECT_TRUE(m_errorHandlerCalled);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareInterface_ReturnValue)
+{
+    const auto interfacePortData =
+        m_senderRuntime->getMiddlewareInterface(iox::capro::Interfaces::INTERNAL, m_runnableName);
+
+    EXPECT_EQ(std::string("/sender"), interfacePortData->m_processName);
+    EXPECT_EQ(iox::capro::ServiceDescription(0u, 0u, 0u), interfacePortData->m_serviceDescription);
+    EXPECT_EQ(false, interfacePortData->m_toBeDestroyed);
+    EXPECT_EQ(true, interfacePortData->m_doInitialOfferForward);
+}
+
+
+TEST_F(PoshRuntime_test, GetMiddlewareInterface_InterfacelistOverflow)
+{
+    m_errorHandlerCalled = false;
+
+    for (auto i = 0u; i < iox::MAX_INTERFACE_NUMBER + 1u; ++i)
+    {
+        m_senderRuntime->getMiddlewareInterface(iox::capro::Interfaces::INTERNAL);
+    }
+
+    EXPECT_TRUE(m_errorHandlerCalled);
+}
+
+
+TEST_F(PoshRuntime_test, SendMessageToRouDi_ValidMessage)
+{
+    m_sendBuffer << mqMessageTypeToString(MqMessageType::IMPL_INTERFACE) << std::string("/sender")
+                 << static_cast<uint32_t>(iox::capro::Interfaces::INTERNAL) << m_runnableName;
+
+    const auto status = m_senderRuntime->sendMessageToRouDi(m_sendBuffer);
+
+    EXPECT_EQ(true, status);
+}
+
+
+TEST_F(PoshRuntime_test, SendMessageToRouDi_InvalidMessage)
+{
+    m_sendBuffer << mqMessageTypeToString(MqMessageType::IMPL_INTERFACE) << std::string()
+                 << static_cast<uint32_t>(iox::capro::Interfaces::INTERNAL) << m_invalidRunnableName;
+
+    const auto status = m_senderRuntime->sendMessageToRouDi(m_sendBuffer);
```

Comment:
could you check if `PoshRuntime::sendMessageToRouDi` is used somewhere? My gut feelings tell me we could remove this function from the runtime

Reply:
yeah, its not used

Reply:
could you then please remove it from PoshRuntime?